src/core_plugins/kibana/public/home/components/home_app.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Home } from './home';
+import { FeatureDirectory } from './feature_directory';
+import {
+  HashRouter as Router,
+  Switch,
+  Route
+} from 'react-router-dom';
+
+export function HomeApp({ addBasePath, directories }) {
+  return (
+    <Router>
+      <Switch>
+        <Route
+          path="/home/feature_directory"
+        >
+          <FeatureDirectory
+            addBasePath={addBasePath}
+            directories={directories}
+          />
+        </Route>
+        <Route
+          path="/"
+        >
+          <Home
+            addBasePath={addBasePath}
+            directories={directories}
+          />
+        </Route>
+      </Switch>
+    </Router>
+  );
+}
+
+HomeApp.propTypes = {
+  addBasePath: PropTypes.func.isRequired,
+  directories: PropTypes.arrayOf(PropTypes.shape({
```

Comment:
I wonder if there is a way to refactor these shapes out.  I'm curious because I'm running into something similar and want to do like a custom jsDoc type rather than redefine the shape everywhere.  Maybe I'll ask around to see if anyone has any ideas.   (no changes required, just thinking out loud).